### PR TITLE
EASY-2614: ISNI regexp allowed 17 digits if the last is an X 

### PR DESCRIPTION
**Pull request by jo-pol**
Fixes EASY-2614: ISNI regexp allowed 17 digits if the last is an X 

#### When applied it will
* test for ISNI edge cases
* some other tests from `easy-validate-dans-bag`
* 

#### Related pull requests on github
* [x] depends on https://github.com/DANS-KNAW/easy-schema/pull/80

### CODE REVIEW

File: src/test/scala/nl/knaw/dans/easy/schemaExamples/DdmSchemaSpec.scala [hunk 1]
```diff
@@ -15,18 +15,42 @@
...
+    validateWithLocal(modify(
+      ".*</ddm:relation>.*",
+      """<ddm:relation scheme="id-type:DOI" href="https://doi.org/42">42</ddm:relation>"""
+    )) shouldBe a[Failure[_]]
+  }
+
+  it should "report an XSS attack" in pendingUntilFixed {
+    validateWithLocal(modify(
+      ".*</ddm:relation>.*",
+      """<ddm:relation xml:lang="nld" href="javascript:alert('XSS')">xxx</ddm:relation>"""
```
Comment (jo-pol) [inline src/test/scala/nl/knaw/dans/easy/schemaExamples/DdmSchemaSpec.scala:93]: Pending as this test is actually implemented as a rule by https://github.com/DANS-KNAW/easy-validate-dans-bag/pull/74
but it could possibly also (partly) done with a regexp.

File: pom.xml [hunk 2]
```diff
@@ -34,7 +34,7 @@
     <inceptionYear>2019</inceptionYear>
 
     <properties>
-        <easy.schema.version>3.3.1</easy.schema.version>
+        <easy.schema.version>3.3.2-SNAPSHOT</easy.schema.version>
```
Comment (jo-pol) [inline pom.xml]: To be fixed once easy-schema is merged and available via maven. That will fix the build.
